server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -513,18 +525,37 @@ private CompletionStage<Result> renderErrorOrRedirectToNextBlock(
               .createAddressQuestion();
 
       if (addressQuestion.needsAddressCorrection()) {
-        return applicantService
-            .getAddressSuggestionGroup(thisBlockUpdated)
-            .thenApplyAsync(
-                addressSuggestionGroup -> {
-                  ImmutableList<AddressSuggestion> suggestions =
-                      addressSuggestionGroup.getAddressSuggestions();
-                  String json = addressSuggestionJsonSerializer.serialize(suggestions);
-
-                  Boolean isEligibilityEnabledOnThisBlock =
-                      thisBlockUpdated.getLeafAddressNodeServiceAreaIds().isPresent();
-
-                  return ok(addressCorrectionBlockView.render(
+        AddressSuggestionGroup addressSuggestionGroup =
+            applicantService
+                .getAddressSuggestionGroup(thisBlockUpdated)
+                .toCompletableFuture()
+                .join();
+        ImmutableList<AddressSuggestion> suggestions =
+            addressSuggestionGroup.getAddressSuggestions();
+
+        AddressSuggestion[] suggestionMatch =
+            suggestions.stream()
+                .filter(suggestion -> suggestion.getAddress().equals(addressQuestion.getAddress()))
+                .toArray(AddressSuggestion[]::new);
+
+        if (suggestionMatch.length > 0) {
+          return confirmAddressWithSuggestions(
+              request,
+              applicantId,
+              programId,
+              blockId,
+              inReview,
+              suggestionMatch[0].getSingleLineAddress(),
+              suggestions);
+        } else {
+          String json = addressSuggestionJsonSerializer.serialize(suggestions);
+
+          Boolean isEligibilityEnabledOnThisBlock =
+              thisBlockUpdated.getLeafAddressNodeServiceAreaIds().isPresent();
+
+          return supplyAsync(
```

Comment:
If we don't find a matching suggestion, we want to show the user the address correction screen.